app/controllers/api/v1/deploys_controller.rb

```diff
@@ -0,0 +1,48 @@
+class Api::V1::DeploysController < Api::V1::ApplicationController
+  include CurrentProject
+
+  before_filter :authorize_project_deployer!
```

Comment:
now we are copy-pasting permission logic .... also the reference logic looks very similar ...